cerberus-api-tests/src/test/groovy/com/nike/cerberus/api/NegativeUserPermissionsApiTests.groovy

```diff
@@ -0,0 +1,196 @@
+package com.nike.cerberus.api
+
+import com.fieldju.commons.PropUtils
+import com.nike.cerberus.api.util.TestUtils
+import com.thedeanda.lorem.Lorem
+import io.restassured.path.json.JsonPath
+import org.apache.commons.lang3.StringUtils
+import org.apache.http.HttpStatus
+import org.testng.annotations.AfterTest
+import org.testng.annotations.BeforeTest
+import org.testng.annotations.Test
+
+import static com.nike.cerberus.api.CerberusCompositeApiActions.*
+import static com.nike.cerberus.api.CerberusApiActions.*
+import static com.nike.cerberus.api.util.TestUtils.generateRandomSdbDescription
+import static com.nike.cerberus.api.util.TestUtils.generateSdbJson
+
+class NegativeUserPermissionsApiTests {
+
+    private static final String PERMISSION_DENIED_JSON_SCHEMA = "$NEGATIVE_JSON_SCHEMA_ROOT_PATH/permission-denied-invalid-auth-token-error.json"
+
+    private String accountId
+    private String roleName
+    private String region
+    private String iamAuthToken
+
+    private String username
+    private String password
+    private String otpDeviceId
+    private String otpSecret
+    private String[] userGroups
+    private String userAuthToken
+    private Map userAuthData
+
+    private Map roleMap
+
+    private def userReadOnlySdb
+    private def userWriteOnlySdb
+
+    private void loadRequiredEnvVars() {
```

Comment:
This is test code, so I'm not too concerned about it, but it does seem like this could be done nicer, more like production code.

E.g. loadRequireEnvVars() could be BeforeClass rather than BeforeTest, these variables could be final at the class level, better yet, the configuration could move into a Config class where it could be more encapsulated.

Reply:
I agree. I just followed the current model of code for now.

I plan to make a separate PR to clean up some of the test code. I will keep your suggestions in mind when I do.

I updated the README.